### PR TITLE
support for providing upload_url in scan_file()

### PR DESCRIPTION
**Pull request by joarleymoraes**

### CODE REVIEW

**Comment by MrRedAmber** [comment]
Looking into this 🤓

File: .gitignore [hunk 1]
```diff
@@ -94,3 +94,4 @@ Icon
 .Trashes
 
 OUT
+venv
```
Comment (MrRedAmber) [inline .gitignore]: Don't change the .gitignore to support your venv configuration. Store it outside of the project folder.

**Comment by MrRedAmber** [comment]
Can you explain to me why this PR is necessary? I'm having trouble finding out a use case scenario where this would be useful. The only two endpoints that accept file uploads are `file/scan` and `file/rescan` and both of these endpoints are implemented in this API.

**Comment by KadenLNelson** [comment]
Closing this PR for inactivity. Please feel free to reopen if you address the comments above.

**Pull request closed without merging**